### PR TITLE
PLT-8100: Add escrow example

### PR DESCRIPTION
**Pull request by sjoerdvisscher**
Pre-submit checklist:
- Branch
    - [x] Tests are provided (if possible)
    - [x] Commit sequence broadly makes sense and have useful messages
    - [ ] Important changes are reflected in changelog.d of the affected packages
    - [x] Relevant tickets are mentioned in commit messages
- PR
    - [ ] (For external contributions) Corresponding issue exists and is linked in the description
    - [x] Targeting main unless this is a cherry-pick backport
    - [x] Self-reviewed the diff
    - [x] Useful pull request description
    - [ ] If relevant, reference the ADR in the PR and reference the PR in the ADR
    - [ ] Reviewer requested

### CODE REVIEW

File: cardano-node-emulator/test/Plutus/Examples/Escrow.hs [hunk 1]
```diff
@@ -0,0 +1,485 @@
...
+  | RefundFailed
+  deriving stock (Show)
+
+makeClassyPrisms ''EscrowError
+
+{- $escrow
+The escrow contract implements the exchange of value between multiple
+parties. It is defined by a list of targets (public keys and script
+addresses, each associated with a value). It works similar to the
+crowdfunding contract in that the contributions can be made independently,
```
Comment (berewt) [inline cardano-node-emulator/test/Plutus/Examples/Escrow.hs:105]: Do we have a crowdfunding example already or do we need to revisit the description?

File: cardano-node-emulator/test/Plutus/Examples/Escrow.hs [hunk 2]
```diff
@@ -0,0 +1,485 @@
...
+has passed. This is because we're assuming that the participants in the
+escrow contract will make their deposits as quickly as possible after
+agreeing on a deal)
+
+The contract supports two modes of operation, manual and automatic. In
+manual mode, all actions are driven by endpoints that exposed via 'payEp'
+'redeemEp' and 'refundEp'. In automatic mode, the 'pay', 'redeem' and
+'refund'actions start immediately. This mode is useful when the escrow is
+called from within another contract, for example during setup (collection of
+the initial deposits).
```
Comment (berewt) [inline cardano-node-emulator/test/Plutus/Examples/Escrow.hs:120]: Does it still make sense?

File: cardano-node-emulator/test/Plutus/Examples/EscrowSpec.hs [hunk 3]
```diff
@@ -0,0 +1,493 @@
...
+    --     ( assertDone con (Trace.walletInstanceTag w3) (const True) "escrow redeem not done"
+    --       .&&. walletFundsChange w1 (Value.adaValueOf (-10))
+    --       .&&. walletFundsChange w2 (Value.adaValueOf 10)
+    --       .&&. walletFundsChange w3 mempty
+    --     )
+    --     redeemTrace
+
+    -- , checkPredicateOptions options "can redeem even if more money than required has been paid in"
+
+    --       -- in this test case we pay in a total of 40 lovelace (10 more than required), for
```
Comment (berewt) [inline cardano-node-emulator/test/Plutus/Examples/EscrowSpec.hs:380]: Can we uncomment it in the future or should we just delete it? (same question for the other example below).

**Review by berewt — APPROVED** [review]
LGTM.

File: cardano-node-emulator/test/Plutus/Examples/Escrow.hs (hunk 1 above)
Comment (sjoerdvisscher) [reply cardano-node-emulator/test/Plutus/Examples/Escrow.hs:105]: We will have one.

File: cardano-node-emulator/test/Plutus/Examples/Escrow.hs (hunk 2 above)
Comment (sjoerdvisscher) [reply cardano-node-emulator/test/Plutus/Examples/Escrow.hs]: No.

File: cardano-node-emulator/test/Plutus/Examples/EscrowSpec.hs (hunk 3 above)
Comment (sjoerdvisscher) [reply cardano-node-emulator/test/Plutus/Examples/EscrowSpec.hs:380]: I wanted to have this visible in one commit, so it is visible what we don't support. I will remove it in the next PR.

File: cardano-node-emulator/src/Cardano/Node/Emulator/API.hs [hunk 4]
```diff
@@ -152,6 +161,16 @@ nextSlot = do
...
+currentTimeRange :: (MonadEmulator m) => m (POSIXTime, POSIXTime)
+currentTimeRange = do
+  slotConfig <- asks pSlotConfig
+  slot <- currentSlot
+  pure
+    ( slotToBeginPOSIXTime slotConfig slot
+    , slotToEndPOSIXTime slotConfig slot
+    )
+
 -- | Call `nextSlot` until the current slot number equals or exceeds the given slot number.
```
Comment (brendanrbrown) [inline cardano-node-emulator/src/Cardano/Node/Emulator/API.hs:174]: Not among your changes for this PR, so feel free to ignore this. 

I noticed the comment is slightly inaccurate: `nextSlot` will be called until the current slot number *exceeds* the given slot number. If they are equal, `pure ()` is called.

**Review by brendanrbrown — APPROVED** [review]
Seems good to me. Thanks for the contribution.

**Pull request merged**